service/src/main/java/uk/gov/mca/beacons/api/accountholder/application/AccountHolderService.java

```diff
@@ -58,13 +67,66 @@ public Optional<AccountHolder> getAccountHolderByAuthId(String authId) {
   public Optional<AccountHolder> updateAccountHolder(
     AccountHolderId id,
     AccountHolder accountHolderUpdate
+  ) {
+    try {
+      log.info(
+        "HELLO Account holder authId is " + accountHolderUpdate.getAuthId()
+      );
+
+      Optional<User> accountHolderInAzure = getAccountHolderFromAzureAd(
+        accountHolderUpdate.getAuthId()
+      );
+
+      if (accountHolderInAzure.isEmpty()) {
+        return Optional.empty();
```

Comment:
Should be throwing an error if it doesn't exist (as this should never happen), not quietly continuing

Reply:
If the user doesn't exist in Azure it will throw an exception and 404 not found error; will add another exception here just in case

Reply:
Yes indeedy! This is because I've put @Transactional on the methods that invoke the repository method and on the the repository itself